axelrod/strategies/lookerup.py

```diff
@@ -0,0 +1,150 @@
+from axelrod import Player, Actions
+import itertools
+
+C, D = Actions.C, Actions.D
+
+class LookerUp(Player):     
+    """     
+    A strategy that uses a lookup table to
+    decide what to do based on a combination of the last m rounds and the opponent's
+    opening n moves. If there isn't enough history to do this (i.e. for the first m
+    rounds) then cooperate.
+
+    The lookup table is implemented as a dict. The keys are 3-tuples giving the
+    opponents first n moves, self's last m moves, and opponents last m moves,
+    all as strings. The values are the moves to play on this round.  For
+    example, if
+
+    - the opponent started by playing C  
+    - my last move was a C the opponents
+    - last move was a D
+
+    the corresponding key would be
+
+        ('C', 'C', 'D')
+
+    and the value would contain the move to play on this turn.
+
+    Some well-known strategies can be expressed as special cases; for example
+    Cooperator is given by the dict:
+
+        {('', '', '') : 'C'}
+
+    where m and n are both zero. Tit For Tat is given by:
+
+    { 
+        ('', 'C', 'D') : 'D',         
+        ('', 'D', 'D') : 'D',         
+        ('', 'C', 'C') : 'C',
+        ('', 'D', 'C') : 'C',     
+    }
+
+    where m=1 and n=0.
+
+    Lookup tables where the move depends on the opponent's first moves (as
+    opposed to most recent moves) will have a non-empty first string in the
+    tuple. For example, this fragment of a dict:
+
+    {          
+        ...         
+        ('C', 'C', 'C') : 'C'.
+        ('D', 'C', 'C') : 'D',
+        ...            
+    }
+
+    states that if self and opponent both cooperated on the previous turn, we
+    should cooperate this turn unless the opponent started by defecting, in
+    which case we should defect.
+
```

Comment:
Could you repeat here what you have written below: 'in the case where no lookup table is passed, the default is the tit for tat...' <- better written better :)